packages/oauth-providers/src/providers/discord/discordAuth.ts

```diff
@@ -30,14 +31,6 @@ export function discordAuth(options: {
       },
     })
 
-    // Avoid CSRF attack by checking state
```

Comment:
To my understanding the state-based CSRF prevention should check the request with Authorization Code, not the request get redirected to `302 https://the-auth-server.com/oauth`. So moving this check later should be safe. Same for other providers.

Previously it could even be a bug? If a guest visits path `/callback?unrelated_query` I image the oauth flow would fail to start.